src/plain.jl

```diff
@@ -2300,6 +2300,29 @@ function hiding_errors(f)
     res
 end
 
+"""
+    get_datasets(file::HDF5File) -> datasets::Vector{HDF5Dataset}
+
+get all datasets in the file without loading the data
+"""
+function get_datasets(file::HDF5File)
+    list = Vector{HDF5Dataset}()
+    get_datasets!(file, list)
```

Comment:
It's typical Julia convention to have the mutated variable appear first (e.g. `get_datasets!(list, file)`).

(not that it matters for this PR since the function is not exported)